src/botcity/maestro/sdk.ts

```diff
@@ -306,4 +306,33 @@ export class BotMaestroSdk {
     await fs.promises.writeFile(filepath, response.data)
     return response.data
   }
+
+  @ensureAccessToken
+  @catchError
+  async getCredential (
+    label: string,
+    key: string
+  ): Promise<any> {
+    const url = `${this._server}/api/v2/credential/${label}/key/${key}`
+    const response: AxiosResponse = await axios.get(url, this.headers).catch((error: any) => {
+      console.log({ error })
+      throw new Error(error.response.data.message)
+    })
+    return response.data
+  }
+
+  @ensureAccessToken
+  @catchError
+  async createCredential (
```

Comment:
We need to check if the top-level object exists using the `label` first and later we can add the `key/value` into it.
Maybe a GET for `/api/v2/credential/{label}` will let us know if it already exists. And if it doesn't, we can create it before adding the new credential item.